### ISSUE TITLE
irshow: compactpending nodes 

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#47288]

**Comment by maleadt** [comment]
So display of `IncrementalCompact::new_new_nodes`? Isn't that handled by the current `new_nodes` iterator? https://github.com/JuliaLang/julia/blob/b9da8a940220d4d21ac2680194435f5078faa167/base/compiler/ssair/show.jl#L755-L760

**Comment by Keno** [comment]
Yes, those. Hmm, that does look like it should be handled - maybe I'm wrong and there was some other issue.

**Comment by maleadt** [comment]
Ah, that's not displaying pending nodes though (`compact.pending_nodes`, when inserting in the uncompacted region). Is there a reason this doesn't just modify `compact.ir.new_nodes`?

**Comment by Keno** [comment]
Ah, sorry, I confused myself. Yes, the issue is `compact.pending_nodes`. You could modify compact.ir.new_nodes, but IncrementalCompact does a sort of the `new_nodes` at the start of the compaction. I guess you could update that sorted list as well, but that's not what it does right now.
